### ISSUE TITLE
Outdated information in "Your first 2D shader"?

### ISSUE THREAD

**Issue by thiagola92** [godotengine/godot-docs#7458]
**Your Godot version:** v4.0.3.rc2.official [2ac4e3bb3]

**Issue description:**  
In the section "Using `TEXTURE` built-in" mentions that i need to read the texture, otherwise i will have an all white rectangle.  
After testing I noticed that didn't made difference, is this right?  

```
void fragment(){
  //this shader will result in an all white rectangle
  COLOR.b = 1.0;
}
```

```
void fragment(){
  COLOR = texture(TEXTURE, UV); //read from texture
  COLOR.b = 1.0; //set blue channel to 1.0
}
```

**URL to the documentation page:** https://docs.godotengine.org/en/stable/tutorials/shaders/your_first_shader/your_first_2d_shader.html#using-texture-built-in

**Comment by jynus** [comment]
I confirm that page needs updating- the new behaviour seems more reasonable to me, but the docs need to reflect this correctly: 
![Same shader in 3.x vs 4.x](https://github.com/godotengine/godot-docs/assets/8576860/a10b5840-920f-43e8-962d-e90d3b281ebb)

**Comment by jynus** [comment]
This is true on Nvidia (as tested), but I am not 100% this is the case on other cards/render hw, such as integrated graphics or software acceleration. I've seen those behave differently on uninitiated data on CPU- can someone check if it works the same for Intel/AMD integrated graphics/Mesa?

**Comment by thiagola92** [comment]
@jynus 
CPU: Intel i5-7600K (4) @ 4.200GHz
GPU: NVIDIA GeForce GTX 1060 6GB

**Comment by skyace65** [comment]
This was fixed by #8583.
